impeller/renderer/backend/gles/device_buffer_gles.cc

```diff
@@ -117,9 +117,17 @@ bool DeviceBufferGLES::BindAndUploadDataIfNecessary(BindingType type) const {
   }
 
   if (dirty_range_.has_value()) {
-    auto range = dirty_range_.value();
-    gl.BufferSubData(target_type, range.offset, range.length,
-                     backing_store_->GetBuffer() + range.offset);
+    Range range = dirty_range_.value();
+    if (gl.MapBufferRange.IsAvailable()) {
+      void* data =
+          gl.MapBufferRange(target_type, range.offset, range.length,
```

Comment:
So at least locally for me, this is the trick that is letting us incrementally write to the device buffers like we do for vulkan/metal. I believe this will be faster but it likely needs a run through the devicelab to confirm.